File: qt/network/gen_qrestaccessmanager.cpp
// +build ignore

#include <QByteArray>
#include <QHttpMultiPart>
#include <QIODevice>
#include <QJsonDocument>
#include <QMap>
#include <QMetaObject>
#include <QNetworkAccessManager>
#include <QNetworkReply>
#include <QNetworkRequest>
#include <QObject>
#include <QRestAccessManager>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVariant>
#include <qrestaccessmanager.h>
#include "gen_qrestaccessmanager.h"
class MiqtVirtualQRestAccessManager : public virtual QRestAccessManager {
public:
MiqtVirtualQRestAccessManager(QNetworkAccessManager* manager): QRestAccessManager(manager) {};
MiqtVirtualQRestAccessManager(QNetworkAccessManager* manager, QObject* parent): QRestAccessManager(manager, parent) {};

virtual ~MiqtVirtualQRestAccessManager() = default;
// cgo.Handle value for overwritten implementation
	intptr_t handle__MetaObject = 0;
// Subclass to allow providing a Go implementation
	virtual const QMetaObject* metaObject() const override {
if (handle__MetaObject == 0) {
return QRestAccessManager::metaObject();
}
QMetaObject* callback_return_value = miqt_exec_callback_QRestAccessManager_MetaObject(const_cast<MiqtVirtualQRestAccessManager*>(this), handle__MetaObject);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	QMetaObject* virtualbase_MetaObject() const {

		return (QMetaObject*) QRestAccessManager::metaObject();

	}
// cgo.Handle value for overwritten implementation
	intptr_t handle__Metacast = 0;
// Subclass to allow providing a Go implementation
	virtual void* qt_metacast(const char* param1) override {
if (handle__Metacast == 0) {
return QRestAccessManager::qt_metacast(param1);
}
const char* sigval1 = (const char*) param1;

		void* callback_return_value = miqt_exec_callback_QRestAccessManager_Metacast(this, handle__Metacast, sigval1);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	void* virtualbase_Metacast(const char* param1) {

		return QRestAccessManager::qt_metacast(param1);

	}
};
QRestAccessManager* QRestAccessManager_new(QNetworkAccessManager* manager) {
return new MiqtVirtualQRestAccessManager(manager);
}
QRestAccessManager* QRestAccessManager_new2(QNetworkAccessManager* manager, QObject* parent) {
return new MiqtVirtualQRestAccessManager(manager, parent);
}
void QRestAccessManager_virtbase(QRestAccessManager* src
, QObject** outptr_QObject
) {
*outptr_QObject = static_cast<QObject*>(src);
}
QMetaObject* QRestAccessManager_MetaObject(const QRestAccessManager* self) {
	return (QMetaObject*) self->metaObject();
}
void* QRestAccessManager_Metacast(QRestAccessManager* self, const char* param1) {
	return self->qt_metacast(param1);
}
struct miqt_string QRestAccessManager_Tr(const char* s) {
	QString _ret = QRestAccessManager::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
QNetworkAccessManager* QRestAccessManager_NetworkAccessManager(const QRestAccessManager* self) {
	return self->networkAccessManager();
}
QNetworkReply* QRestAccessManager_DeleteResource(QRestAccessManager* self, QNetworkRequest* request) {
	return self->deleteResource(*request);
}
QNetworkReply* QRestAccessManager_Head(QRestAccessManager* self, QNetworkRequest* request) {
	return self->head(*request);
}
QNetworkReply* QRestAccessManager_Get(QRestAccessManager* self, QNetworkRequest* request) {
	return self->get(*request);
}
QNetworkReply* QRestAccessManager_Get2(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string data) {
	QByteArray data_QByteArray(data.data, data.len);
	return self->get(*request, data_QByteArray);
}
QNetworkReply* QRestAccessManager_Get3(QRestAccessManager* self, QNetworkRequest* request, QJsonDocument* data) {
	return self->get(*request, *data);
}
QNetworkReply* QRestAccessManager_Get4(QRestAccessManager* self, QNetworkRequest* request, QIODevice* data) {
	return self->get(*request, data);
}
QNetworkReply* QRestAccessManager_Post(QRestAccessManager* self, QNetworkRequest* request, QJsonDocument* data) {
	return self->post(*request, *data);
}
QNetworkReply* QRestAccessManager_Post2(QRestAccessManager* self, QNetworkRequest* request, struct miqt_map /* of struct miqt_string to QVariant* */  data) {
	QVariantMap data_QMap;
	struct miqt_string* data_karr = static_cast<struct miqt_string*>(data.keys);
	QVariant** data_varr = static_cast<QVariant**>(data.values);
	for(size_t i = 0; i < data.len; ++i) {
		QString data_karr_i_QString = QString::fromUtf8(data_karr[i].data, data_karr[i].len);
		data_QMap[data_karr_i_QString] = *(data_varr[i]);
	}
	return self->post(*request, data_QMap);
}
QNetworkReply* QRestAccessManager_Post3(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string data) {
	QByteArray data_QByteArray(data.data, data.len);
	return self->post(*request, data_QByteArray);
}
QNetworkReply* QRestAccessManager_Post4(QRestAccessManager* self, QNetworkRequest* request, QHttpMultiPart* data) {
	return self->post(*request, data);
}
QNetworkReply* QRestAccessManager_Post5(QRestAccessManager* self, QNetworkRequest* request, QIODevice* data) {
	return self->post(*request, data);
}
QNetworkReply* QRestAccessManager_Put(QRestAccessManager* self, QNetworkRequest* request, QJsonDocument* data) {
	return self->put(*request, *data);
}
QNetworkReply* QRestAccessManager_Put2(QRestAccessManager* self, QNetworkRequest* request, struct miqt_map /* of struct miqt_string to QVariant* */  data) {
	QVariantMap data_QMap;
	struct miqt_string* data_karr = static_cast<struct miqt_string*>(data.keys);
	QVariant** data_varr = static_cast<QVariant**>(data.values);
	for(size_t i = 0; i < data.len; ++i) {
		QString data_karr_i_QString = QString::fromUtf8(data_karr[i].data, data_karr[i].len);
		data_QMap[data_karr_i_QString] = *(data_varr[i]);
	}
	return self->put(*request, data_QMap);
}
QNetworkReply* QRestAccessManager_Put3(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string data) {
	QByteArray data_QByteArray(data.data, data.len);
	return self->put(*request, data_QByteArray);
}
QNetworkReply* QRestAccessManager_Put4(QRestAccessManager* self, QNetworkRequest* request, QHttpMultiPart* data) {
	return self->put(*request, data);
}
QNetworkReply* QRestAccessManager_Put5(QRestAccessManager* self, QNetworkRequest* request, QIODevice* data) {
	return self->put(*request, data);
}
QNetworkReply* QRestAccessManager_Patch(QRestAccessManager* self, QNetworkRequest* request, QJsonDocument* data) {
	return self->patch(*request, *data);
}
QNetworkReply* QRestAccessManager_Patch2(QRestAccessManager* self, QNetworkRequest* request, struct miqt_map /* of struct miqt_string to QVariant* */  data) {
	QVariantMap data_QMap;
	struct miqt_string* data_karr = static_cast<struct miqt_string*>(data.keys);
	QVariant** data_varr = static_cast<QVariant**>(data.values);
	for(size_t i = 0; i < data.len; ++i) {
		QString data_karr_i_QString = QString::fromUtf8(data_karr[i].data, data_karr[i].len);
		data_QMap[data_karr_i_QString] = *(data_varr[i]);
	}
	return self->patch(*request, data_QMap);
}
QNetworkReply* QRestAccessManager_Patch3(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string data) {
	QByteArray data_QByteArray(data.data, data.len);
	return self->patch(*request, data_QByteArray);
}
QNetworkReply* QRestAccessManager_Patch4(QRestAccessManager* self, QNetworkRequest* request, QIODevice* data) {
	return self->patch(*request, data);
}
QNetworkReply* QRestAccessManager_SendCustomRequest(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string method, struct miqt_string data) {
	QByteArray method_QByteArray(method.data, method.len);
	QByteArray data_QByteArray(data.data, data.len);
	return self->sendCustomRequest(*request, method_QByteArray, data_QByteArray);
}
QNetworkReply* QRestAccessManager_SendCustomRequest2(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string method, QIODevice* data) {
	QByteArray method_QByteArray(method.data, method.len);
	return self->sendCustomRequest(*request, method_QByteArray, data);
}
QNetworkReply* QRestAccessManager_SendCustomRequest3(QRestAccessManager* self, QNetworkRequest* request, struct miqt_string method, QHttpMultiPart* data) {
	QByteArray method_QByteArray(method.data, method.len);
	return self->sendCustomRequest(*request, method_QByteArray, data);
}
struct miqt_string QRestAccessManager_Tr2(const char* s, const char* c) {
	QString _ret = QRestAccessManager::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
struct miqt_string QRestAccessManager_Tr3(const char* s, const char* c, int n) {
	QString _ret = QRestAccessManager::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QRestAccessManager_override_virtual_MetaObject(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQRestAccessManager*>( (QRestAccessManager*)(self) )->handle__MetaObject = slot;
}
QMetaObject* QRestAccessManager_virtualbase_MetaObject(const void* self) {
	return ( (const MiqtVirtualQRestAccessManager*)(self) )->virtualbase_MetaObject();
}
void QRestAccessManager_override_virtual_Metacast(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQRestAccessManager*>( (QRestAccessManager*)(self) )->handle__Metacast = slot;
}
void* QRestAccessManager_virtualbase_Metacast(void* self, const char* param1) {
	return ( (MiqtVirtualQRestAccessManager*)(self) )->virtualbase_Metacast(param1);
}
void QRestAccessManager_Delete(QRestAccessManager* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQRestAccessManager*>( self );
	} else {
		delete self;
	}
}

File: qt/network/gen_qrestaccessmanager.go
package network

import (
	"github.com/mappu/miqt/qt"
	"unsafe"
)

type QRestAccessManager struct {
	h          uintptr
	isSubclass bool
}

// NewQRestAccessManager constructs a new QRestAccessManager object.
func NewQRestAccessManager(manager *QNetworkAccessManager) *QRestAccessManager {

	ret := newQRestAccessManager(QRestAccessManager_new(manager.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQRestAccessManager2 constructs a new QRestAccessManager object.
func NewQRestAccessManager2(manager *QNetworkAccessManager, parent *qt.QObject) *QRestAccessManager {

	ret := newQRestAccessManager(QRestAccessManager_new2(manager.cPointer(), (*QObject)(parent.UnsafePointer())))
	ret.isSubclass = true
	return ret
}

func (this *QRestAccessManager) MetaObject() *qt.QMetaObject {
	return qt.UnsafeNewQMetaObject(unsafe.Pointer(QRestAccessManager_MetaObject(this.h)))
}

func (this *QRestAccessManager) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QRestAccessManager_Metacast(this.h, param1_Cstring))
}

func QRestAccessManager_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QRestAccessManager_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QRestAccessManager) NetworkAccessManager() *QNetworkAccessManager {
	return newQNetworkAccessManager(QRestAccessManager_NetworkAccessManager(this.h))
}

func (this *QRestAccessManager) DeleteResource(request *QNetworkRequest) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_DeleteResource(this.h, request.cPointer()))
}

func (this *QRestAccessManager) Head(request *QNetworkRequest) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Head(this.h, request.cPointer()))
}

func (this *QRestAccessManager) Get(request *QNetworkRequest) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Get(this.h, request.cPointer()))
}

func (this *QRestAccessManager) Get2(request *QNetworkRequest, data []byte) *QNetworkReply {
	data_alias := struct_miqt_string{}
	data_alias.data = (char)(unsafe.Pointer(&data[0]))
	data_alias.len = size_t(len(data))
	return newQNetworkReply(QRestAccessManager_Get2(this.h, request.cPointer(), data_alias))
}

func (this *QRestAccessManager) Get3(request *QNetworkRequest, data *qt.QJsonDocument) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Get3(this.h, request.cPointer(), (*QJsonDocument)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Get4(request *QNetworkRequest, data *qt.QIODevice) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Get4(this.h, request.cPointer(), (*QIODevice)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Post(request *QNetworkRequest, data *qt.QJsonDocument) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Post(this.h, request.cPointer(), (*QJsonDocument)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Post2(request *QNetworkRequest, data map[string]qt.QVariant) *QNetworkReply {
	data_Keys_CArray := (*[0xffff]struct_miqt_string)(malloc(size_t(int(unsafe.Sizeof(struct_miqt_string{})) * len(data))))
	defer free(unsafe.Pointer(data_Keys_CArray))
	data_Values_CArray := (*[0xffff]*QVariant)(malloc(size_t(8 * len(data))))
	defer free(unsafe.Pointer(data_Values_CArray))
	data_ctr := 0
	for data_k, data_v := range data {
		data_k_ms := struct_miqt_string{}
		data_k_ms.data = CString(data_k)
		data_k_ms.len = size_t(len(data_k))
		defer free(unsafe.Pointer(data_k_ms.data))
		data_Keys_CArray[data_ctr] = data_k_ms
		data_Values_CArray[data_ctr] = (*QVariant)(data_v.UnsafePointer())
		data_ctr++
	}
	data_mm := struct_miqt_map{
		len:    size_t(len(data)),
		keys:   unsafe.Pointer(data_Keys_CArray),
		values: unsafe.Pointer(data_Values_CArray),
	}
	return newQNetworkReply(QRestAccessManager_Post2(this.h, request.cPointer(), data_mm))
}

func (this *QRestAccessManager) Post3(request *QNetworkRequest, data []byte) *QNetworkReply {
	data_alias := struct_miqt_string{}
	data_alias.data = (char)(unsafe.Pointer(&data[0]))
	data_alias.len = size_t(len(data))
	return newQNetworkReply(QRestAccessManager_Post3(this.h, request.cPointer(), data_alias))
}

func (this *QRestAccessManager) Post4(request *QNetworkRequest, data *QHttpMultiPart) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Post4(this.h, request.cPointer(), data.cPointer()))
}

func (this *QRestAccessManager) Post5(request *QNetworkRequest, data *qt.QIODevice) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Post5(this.h, request.cPointer(), (*QIODevice)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Put(request *QNetworkRequest, data *qt.QJsonDocument) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Put(this.h, request.cPointer(), (*QJsonDocument)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Put2(request *QNetworkRequest, data map[string]qt.QVariant) *QNetworkReply {
	data_Keys_CArray := (*[0xffff]struct_miqt_string)(malloc(size_t(int(unsafe.Sizeof(struct_miqt_string{})) * len(data))))
	defer free(unsafe.Pointer(data_Keys_CArray))
	data_Values_CArray := (*[0xffff]*QVariant)(malloc(size_t(8 * len(data))))
	defer free(unsafe.Pointer(data_Values_CArray))
	data_ctr := 0
	for data_k, data_v := range data {
		data_k_ms := struct_miqt_string{}
		data_k_ms.data = CString(data_k)
		data_k_ms.len = size_t(len(data_k))
		defer free(unsafe.Pointer(data_k_ms.data))
		data_Keys_CArray[data_ctr] = data_k_ms
		data_Values_CArray[data_ctr] = (*QVariant)(data_v.UnsafePointer())
		data_ctr++
	}
	data_mm := struct_miqt_map{
		len:    size_t(len(data)),
		keys:   unsafe.Pointer(data_Keys_CArray),
		values: unsafe.Pointer(data_Values_CArray),
	}
	return newQNetworkReply(QRestAccessManager_Put2(this.h, request.cPointer(), data_mm))
}

func (this *QRestAccessManager) Put3(request *QNetworkRequest, data []byte) *QNetworkReply {
	data_alias := struct_miqt_string{}
	data_alias.data = (char)(unsafe.Pointer(&data[0]))
	data_alias.len = size_t(len(data))
	return newQNetworkReply(QRestAccessManager_Put3(this.h, request.cPointer(), data_alias))
}

func (this *QRestAccessManager) Put4(request *QNetworkRequest, data *QHttpMultiPart) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Put4(this.h, request.cPointer(), data.cPointer()))
}

func (this *QRestAccessManager) Put5(request *QNetworkRequest, data *qt.QIODevice) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Put5(this.h, request.cPointer(), (*QIODevice)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Patch(request *QNetworkRequest, data *qt.QJsonDocument) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Patch(this.h, request.cPointer(), (*QJsonDocument)(data.UnsafePointer())))
}

func (this *QRestAccessManager) Patch2(request *QNetworkRequest, data map[string]qt.QVariant) *QNetworkReply {
	data_Keys_CArray := (*[0xffff]struct_miqt_string)(malloc(size_t(int(unsafe.Sizeof(struct_miqt_string{})) * len(data))))
	defer free(unsafe.Pointer(data_Keys_CArray))
	data_Values_CArray := (*[0xffff]*QVariant)(malloc(size_t(8 * len(data))))
	defer free(unsafe.Pointer(data_Values_CArray))
	data_ctr := 0
	for data_k, data_v := range data {
		data_k_ms := struct_miqt_string{}
		data_k_ms.data = CString(data_k)
		data_k_ms.len = size_t(len(data_k))
		defer free(unsafe.Pointer(data_k_ms.data))
		data_Keys_CArray[data_ctr] = data_k_ms
		data_Values_CArray[data_ctr] = (*QVariant)(data_v.UnsafePointer())
		data_ctr++
	}
	data_mm := struct_miqt_map{
		len:    size_t(len(data)),
		keys:   unsafe.Pointer(data_Keys_CArray),
		values: unsafe.Pointer(data_Values_CArray),
	}
	return newQNetworkReply(QRestAccessManager_Patch2(this.h, request.cPointer(), data_mm))
}

func (this *QRestAccessManager) Patch3(request *QNetworkRequest, data []byte) *QNetworkReply {
	data_alias := struct_miqt_string{}
	data_alias.data = (char)(unsafe.Pointer(&data[0]))
	data_alias.len = size_t(len(data))
	return newQNetworkReply(QRestAccessManager_Patch3(this.h, request.cPointer(), data_alias))
}

func (this *QRestAccessManager) Patch4(request *QNetworkRequest, data *qt.QIODevice) *QNetworkReply {
	return newQNetworkReply(QRestAccessManager_Patch4(this.h, request.cPointer(), (*QIODevice)(data.UnsafePointer())))
}

func (this *QRestAccessManager) SendCustomRequest(request *QNetworkRequest, method []byte, data []byte) *QNetworkReply {
	method_alias := struct_miqt_string{}
	method_alias.data = (char)(unsafe.Pointer(&method[0]))
	method_alias.len = size_t(len(method))
	data_alias := struct_miqt_string{}
	data_alias.data = (char)(unsafe.Pointer(&data[0]))
	data_alias.len = size_t(len(data))
	return newQNetworkReply(QRestAccessManager_SendCustomRequest(this.h, request.cPointer(), method_alias, data_alias))
}

func (this *QRestAccessManager) SendCustomRequest2(request *QNetworkRequest, method []byte, data *qt.QIODevice) *QNetworkReply {
	method_alias := struct_miqt_string{}
	method_alias.data = (char)(unsafe.Pointer(&method[0]))
	method_alias.len = size_t(len(method))
	return newQNetworkReply(QRestAccessManager_SendCustomRequest2(this.h, request.cPointer(), method_alias, (*QIODevice)(data.UnsafePointer())))
}

func (this *QRestAccessManager) SendCustomRequest3(request *QNetworkRequest, method []byte, data *QHttpMultiPart) *QNetworkReply {
	method_alias := struct_miqt_string{}
	method_alias.data = (char)(unsafe.Pointer(&method[0]))
	method_alias.len = size_t(len(method))
	return newQNetworkReply(QRestAccessManager_SendCustomRequest3(this.h, request.cPointer(), method_alias, data.cPointer()))
}

func QRestAccessManager_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QRestAccessManager_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QRestAccessManager_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QRestAccessManager_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QRestAccessManager) callVirtualBase_Event(event *qt.QEvent) bool {

	return (bool)(QRestAccessManager_virtualbase_Event(unsafe.Pointer(this.h), (*QEvent)(event.UnsafePointer())))

}
func (this *QRestAccessManager) OnEvent(slot func(super func(event *qt.QEvent) bool, event *qt.QEvent) bool) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_Event(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_Event
func miqt_exec_callback_QRestAccessManager_Event(self QRestAccessManager, cb intptr_t, event *QEvent) bool {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *qt.QEvent) bool, event *qt.QEvent) bool)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQEvent(unsafe.Pointer(event))

	virtualReturn := gofunc((&QRestAccessManager{h: self}).callVirtualBase_Event, slotval1)

	return (bool)(virtualReturn)

}

func (this *QRestAccessManager) callVirtualBase_EventFilter(watched *qt.QObject, event *qt.QEvent) bool {

	return (bool)(QRestAccessManager_virtualbase_EventFilter(unsafe.Pointer(this.h), (*QObject)(watched.UnsafePointer()), (*QEvent)(event.UnsafePointer())))

}
func (this *QRestAccessManager) OnEventFilter(slot func(super func(watched *qt.QObject, event *qt.QEvent) bool, watched *qt.QObject, event *qt.QEvent) bool) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_EventFilter(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_EventFilter
func miqt_exec_callback_QRestAccessManager_EventFilter(self QRestAccessManager, cb intptr_t, watched *QObject, event *QEvent) bool {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(watched *qt.QObject, event *qt.QEvent) bool, watched *qt.QObject, event *qt.QEvent) bool)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQObject(unsafe.Pointer(watched))

	slotval2 := qt.UnsafeNewQEvent(unsafe.Pointer(event))

	virtualReturn := gofunc((&QRestAccessManager{h: self}).callVirtualBase_EventFilter, slotval1, slotval2)

	return (bool)(virtualReturn)

}

func (this *QRestAccessManager) callVirtualBase_TimerEvent(event *qt.QTimerEvent) {

	QRestAccessManager_virtualbase_TimerEvent(unsafe.Pointer(this.h), (*QTimerEvent)(event.UnsafePointer()))

}
func (this *QRestAccessManager) OnTimerEvent(slot func(super func(event *qt.QTimerEvent), event *qt.QTimerEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_TimerEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_TimerEvent
func miqt_exec_callback_QRestAccessManager_TimerEvent(self QRestAccessManager, cb intptr_t, event *QTimerEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *qt.QTimerEvent), event *qt.QTimerEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQTimerEvent(unsafe.Pointer(event))

	gofunc((&QRestAccessManager{h: self}).callVirtualBase_TimerEvent, slotval1)

}

func (this *QRestAccessManager) callVirtualBase_ChildEvent(event *qt.QChildEvent) {

	QRestAccessManager_virtualbase_ChildEvent(unsafe.Pointer(this.h), (*QChildEvent)(event.UnsafePointer()))

}
func (this *QRestAccessManager) OnChildEvent(slot func(super func(event *qt.QChildEvent), event *qt.QChildEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_ChildEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_ChildEvent
func miqt_exec_callback_QRestAccessManager_ChildEvent(self QRestAccessManager, cb intptr_t, event *QChildEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *qt.QChildEvent), event *qt.QChildEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQChildEvent(unsafe.Pointer(event))

	gofunc((&QRestAccessManager{h: self}).callVirtualBase_ChildEvent, slotval1)

}

func (this *QRestAccessManager) callVirtualBase_CustomEvent(event *qt.QEvent) {

	QRestAccessManager_virtualbase_CustomEvent(unsafe.Pointer(this.h), (*QEvent)(event.UnsafePointer()))

}
func (this *QRestAccessManager) OnCustomEvent(slot func(super func(event *qt.QEvent), event *qt.QEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_CustomEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_CustomEvent
func miqt_exec_callback_QRestAccessManager_CustomEvent(self QRestAccessManager, cb intptr_t, event *QEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *qt.QEvent), event *qt.QEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQEvent(unsafe.Pointer(event))

	gofunc((&QRestAccessManager{h: self}).callVirtualBase_CustomEvent, slotval1)

}

func (this *QRestAccessManager) callVirtualBase_ConnectNotify(signal *qt.QMetaMethod) {

	QRestAccessManager_virtualbase_ConnectNotify(unsafe.Pointer(this.h), (*QMetaMethod)(signal.UnsafePointer()))

}
func (this *QRestAccessManager) OnConnectNotify(slot func(super func(signal *qt.QMetaMethod), signal *qt.QMetaMethod)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_ConnectNotify(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_ConnectNotify
func miqt_exec_callback_QRestAccessManager_ConnectNotify(self QRestAccessManager, cb intptr_t, signal *QMetaMethod) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(signal *qt.QMetaMethod), signal *qt.QMetaMethod))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQMetaMethod(unsafe.Pointer(signal))

	gofunc((&QRestAccessManager{h: self}).callVirtualBase_ConnectNotify, slotval1)

}

func (this *QRestAccessManager) callVirtualBase_DisconnectNotify(signal *qt.QMetaMethod) {

	QRestAccessManager_virtualbase_DisconnectNotify(unsafe.Pointer(this.h), (*QMetaMethod)(signal.UnsafePointer()))

}
func (this *QRestAccessManager) OnDisconnectNotify(slot func(super func(signal *qt.QMetaMethod), signal *qt.QMetaMethod)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QRestAccessManager_override_virtual_DisconnectNotify(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QRestAccessManager_DisconnectNotify
func miqt_exec_callback_QRestAccessManager_DisconnectNotify(self QRestAccessManager, cb intptr_t, signal *QMetaMethod) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(signal *qt.QMetaMethod), signal *qt.QMetaMethod))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := qt.UnsafeNewQMetaMethod(unsafe.Pointer(signal))

	gofunc((&QRestAccessManager{h: self}).callVirtualBase_DisconnectNotify, slotval1)

}
